mita/openstack.py

```diff
@@ -211,28 +237,20 @@ def _get_network(self, network: Optional[str] = None) -> OpenStackNetwork:
             else [network]
         )
 
-        _available_networks = self.driver_v2.ex_list_networks()
-
         for net in _networks:
+            # Treating an exception as a soft error as it is possible to find another
+            # suitable network from the list.
             try:
-                os_net = [n for n in _available_networks if n.name == net][0]
+                os_net = self._get_network_by_name(name=net)
 
-                if not self._has_free_ip_address(os_net.id):
-                    logger.debug("%s does not meet the requirements", net)
+                if not self._has_free_ip_address(net=os_net):
+                    logger.debug("%s has no free ip addresses.", os_net.name)
                     continue
 
-                self.subnet = [
-                    s.cidr
-                    for s in self.driver_v2.ex_list_subnets()
-                    if s.id in os_net.extra.get("subnets")
-                ][0]
-
                 return os_net
-            except TypeError:
-                logger.warning("%s is not accessible.", net)
             except BaseException as be:  # noqa
-                logger.warning("Something went wrong during network selection process.")
-                logger.info(be)
+                logger.debug(be)
```

Comment:
Instead of `logger.debug()`, It's appropriate to keep this at the WARNING level with `logger.warning()` here. Ideally we never hit this exception, but if we do, it could indicate a problem on the OpenStack server that we might report to the OpenStack administrators.